### PR TITLE
Makes qdeleted and qdestroying macros

### PR DESCRIPTION
**Pull request by Cyberboss**

### CODE REVIEW

File: code/__DEFINES/qdel.dm [hunk 1]
```diff
@@ -13,3 +13,6 @@
 #define GC_QUEUED_FOR_QUEUING -1
 #define GC_QUEUED_FOR_HARD_DEL -2
 #define GC_CURRENTLY_BEING_QDELETED -3
+
+#define qdeleted(X) (istype(X, /datum) && X.gc_destroyed)
```
Comment (RemieRichards) [inline code/__DEFINES/qdel.dm]: I don't like macros pretending to be procs (non-UPPERCASE macro names)
It should be evident when you use a macro that it's a macro, and we do that via UPPERCASE.

**Review by RemieRichards — APPROVED** [review]

**Comment by Cyberboss** [comment]
Don't believe Travis' lies

**Comment by Cyberboss** [comment]
This actually doesn't make sense

**Comment by MrStonedOne** [comment]
The is type has more overhead then the process call

**Comment by Cyberboss** [comment]
Meh, baby steps I guess...

**Comment by MrStonedOne** [comment]
No, remove the is type. Add a null check. Call it good

**Comment by Cyberboss** [comment]
Travis tho: https://www.youtube.com/watch?v=IdtKbq3Omkw

**Comment by MrStonedOne** [comment]
It should return true if its null thou

Also try reading Travis's error. One of the lines has two calls to qdeleted and you only changed one

**Comment by Cyberboss** [comment]
I trusted you sed...

On Feb 3, 2017 2:13 PM, "Kyle Spier-Swenson" <notifications@github.com>
wrote:

It should return true if its null thou

Also try reading Travis's error. One of the lines has two calls to qdeleted
and you only changed one

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub
<https://github.com/tgstation/tgstation/pull/23666#issuecomment-277335744>,
or mute the thread

**Comment by optimumtact** [comment]
\>travis is lies

**Pull request merged**